core/server/master/src/main/java/alluxio/master/metastore/rocks/RocksStore.java

```diff
@@ -217,6 +226,8 @@ public synchronized void restoreFromCheckpoint(CheckpointInputStream input) thro
   @Override
   public synchronized void close() {
     stopDb();
+    mDbOpts.close();
+    mDbOpts = null;
```

Comment:
My personal preference is to never set a member variable to null unless it is passed in from outside of the object.  It was usually done to prevent obsolete references, but how could this reference be obsolete when this object crated it in the first place..

Reply:
Oracle doc actually mentioned this https://docs.oracle.com/cd/E19159-01/819-3681/abebi/index.html

Setting the var to null helps the GC to tell that this object can be recycled. On 2nd thought in this particular case the `RocksObject` instances typically have the same life cycle as the holder instances themselves. So I don't see need in setting these to null explicitly. 

I will remove these setting to null lines. Also I don't have to remove the `final` and start worrying about thread safety again.